engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ArrowFixedSizeArrayBuilder.java

```diff
@@ -33,9 +31,21 @@ public LogicalLayout getUnit() {
     return unit;
   }
 
+  public boolean isSealed() {
+    return sealed;
+  }
+
+  public ByteBufferDirect getBuffer() {
+    return buffer;
+  }
+
+  public int getSize() {
+    return size;
+  }
+
   @ExportMessage
   public boolean hasArrayElements() {
-    return true;
+    return false;
```

Comment:
You don't have to implement this method at all and then you can remove also associated `writeArrayElement` & co.

Reply:
Weird, I could swear I was getting compilation errors :shrug: